akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowFoldAsyncSpec.scala

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (C) 2014-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.scaladsl
+
+import akka.NotUsed
+
+import scala.concurrent.{ Await, Future }
+import scala.util.control.NoStackTrace
+
+import akka.stream.ActorMaterializer
+import akka.testkit.AkkaSpec
+import akka.stream.testkit.Utils._
+import scala.concurrent.duration._
+
+class FlowFoldAsyncSpec extends AkkaSpec {
+  implicit val materializer = ActorMaterializer()
+  implicit def ec = materializer.executionContext
+
+  "A FoldAsync" must {
+    val input = 1 to 100
+    val expected = input.sum
+    val inputSource = Source(input).filter(_ ⇒ true).map(identity)
+    //TODO:val foldSource = inputSource.fold[Int](0)(_ + _).filter(_ ⇒ true).map(identity)
+    val foldFlow = Flow[Int].filter(_ ⇒ true).map(identity).foldAsync(0) {
+      (a, b) ⇒ Future { Thread.sleep(100); a + b }
+    }.filter(_ ⇒ true).map(identity)
+    //TODO:val foldSink = Sink.fold[Int, Int](0)(_ + _)
+
+    /* TODO:
```

Comment:
If you're up for it, would be lovely if we could get them in in the same PR. WDYT?